tests/framework-tests/core/js/ResourceLoaderTests.js

```diff
@@ -163,4 +269,241 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+            mocks.destroy();
+            jqUnit.start();
+        });
+    });
+
+    /** FLUID-4982: Partially filled out resource blocks **/
+
+    fluid.defaults("fluid.tests.FLUID4982incomplete", {
+        gradeNames: "fluid.tests.FLUID4982base",
+        resources: {
+            initModel: {
+                dataType: "json"
+            }
+        }
+    });
+
+    jqUnit.test("FLUID-4982: Improperly filled resource blocks", function () {
+        jqUnit.expectFrameworkDiagnostic("Framework exception mentioning valid options on incomplete resource block", function () {
+            fluid.tests.FLUID4982incomplete();
+        }, ["resource loader", "url"]);
+    });
+
+    /** FLUID-4982: Overriding primary resourceLoader blocks with promise or dataSource sources */
+
+    fluid.defaults("fluid.tests.fluid4982overrideBase", {
+        gradeNames: ["fluid.modelComponent", "fluid.resourceLoader"],
+        model: {
+            messages: "{that}.resources.messages"
+        },
+        members: {
+            creationPromise: "@expand:fluid.promise()"
+        },
+        listeners: {
+            "onCreate.resolveCreation": "{that}.creationPromise.resolve"
+        },
+        resources: {
+            messages: {
+                dataType: "json",
+                locale: "fr",
+                url: "../data/messages2.json"
+            }
+        },
+        components: {
+            dataSource: {
+                type: "fluid.dataSource.URL",
+                options: {
+                    url: "../data/messages3_en.json"
+                }
+            }
+        }
+    });
+
+    fluid.defaults("fluid.tests.fluid4982overrideDerived", {
+        gradeNames: "fluid.tests.fluid4982overrideBase",
+        resources: {
+            messages: {
+                dataSource: "{that}.dataSource",
+                // Make sure the ResourceLoader does not attempt to parse the DataSource output as JSON which is already parsed
+                dataType: "none"
+            }
+        }
+    });
+
+    fluid.defaults("fluid.tests.fluid4982overridePromise", {
+        gradeNames: "fluid.tests.fluid4982overrideBase",
+        invokers: {
+            fetchMessages: "fluid.tests.fluid4982fetchMessages"
+        },
+        resources: {
+            messages: {
+                promiseFunc: "{that}.fetchMessages",
+                promiseArgs: "../data/messages3_en.json"
+            }
+        }
+    });
+
+    fluid.tests.fluid4982fetchMessages = function (pathname) {
+        return fluid.dataSource.URL.handle.http(null, new URL(pathname, document.location));
+    };
+
+    fluid.tests.fluid4982OverrideFixtures = [{
+        grade: "fluid.tests.fluid4982overrideBase",
+        expected: "Après moi, le déluge!"
+    }, {
+        grade: "fluid.tests.fluid4982overrideDerived",
+        expected: "upper, middle, lower"
+    }, , {
+        grade: "fluid.tests.fluid4982overridePromise",
+        expected: "upper, middle, lower"
```

Comment:
Can you use messages1_en.json and its expected payload for the promise example?   It seems a little cleaner than getting to exactly the same outcome via two mechanisms in test definitions that are right next to one another.